src/compile/scale/parse.ts

```diff
@@ -45,15 +46,20 @@ function parseUnitScaleCore(model: UnitModel): ScaleComponentIndex {
   const mark = model.mark();
 
   return SCALE_CHANNELS.reduce((scaleComponents: ScaleComponentIndex, channel: ScaleChannel) => {
+    // if mark is geoshape scale channel encodes the geojson
+    if (mark === GEOSHAPE && isGeoJSONFieldDef(encoding[channel])) {
+      return scaleComponents;
+    }
+
     let fieldDef: FieldDef<string>;
     let specifiedScale: Scale = {};
 
     const channelDef = encoding[channel];
 
-    if (isFieldDef(channelDef)) {
+    if (isFieldDef(channelDef) && isScaleFieldDef(channelDef)) {
```

Comment:
I dont think its causing the issues. I slightly rewrote it and the issues are the same